app/app.go

```diff
@@ -432,6 +434,21 @@ func New(
 			return nil, err
 		}
 
+		// Parse app_state.app_params.allow_zero_fee_txs and save to auth (acc) store.
+		if raw, ok := genesisState["app_params"]; ok {
+			var appParams struct {
+				AllowZeroFeeTxs bool `json:"allow_zero_fee_txs"`
+			}
+			if err := json.Unmarshal(raw, &appParams); err == nil {
+				store := ctx.KVStore(app.GetKey(authtypes.StoreKey))
+				val := byte(0x00)
+				if appParams.AllowZeroFeeTxs {
+					val = 0x01
+				}
+				store.Set([]byte(appparams.AllowZeroFeeTxsKey), []byte{val})
+			}
+		}
+
```

Comment:
suggestion: IMO we should set the value in the store even if the genesis param is not defined.
```suggestion
		// Parse app_state.app_params.allow_zero_fee_txs and save to auth (acc) store.
		val := byte(0x00)
		if raw, ok := genesisState["app_params"]; ok {
			var appParams struct {
				AllowZeroFeeTxs bool `json:"allow_zero_fee_txs"`
			}
			if err := json.Unmarshal(raw, &appParams); err == nil  && appParams.AllowZeroFeeTxs{
				val = 0x01
			}
		}
		store := ctx.KVStore(app.GetKey(authtypes.StoreKey))
		store.Set([]byte(appparams.AllowZeroFeeTxsKey), []byte{val})



```